src/lib.rs

```diff
@@ -718,11 +876,8 @@ impl Expr {
                     buf.push_str("(?:");
                 }
 
-                let is_empty = |e: &Expr| match e {
-                    Expr::Empty => true,
-                    _ => false,
-                };
-                let contains_empty = children.iter().any(&is_empty);
+                let is_empty = |e: &Expr| matches!(e, Expr::Empty);
```

Comment:
This is the reason the build is failing, `matches!` doesn't exist on Rust 1.32.0 which is the oldest version we currently support in fancy-regex. Mind rewriting it to a match?